src/docs/cookbook/networking/fetch-data.md

```diff
@@ -190,18 +190,22 @@ You must provide two parameters:
 
 Note that `snapshot.hasData` only returns `true`
 when the snapshot contains a non-null data value.
-This is why the `fetchAlbum` function should throw an exception
+
+Because `fetchAlbum` can only return non-null values,
+the function should throw an exception
 even in the case of a "404 Not Found" server response.
-If `fetchAlbum` returns `null`
-then the spinner displays indefinitely.
+Throwing an exception sets the `snapshot.hasError` to `true`
+which can be used to display an error message.
+
+Otherwise, the spinner will be displayed.
 
 <!-- skip -->
 ```dart
 FutureBuilder<Album>(
   future: futureAlbum,
   builder: (context, snapshot) {
     if (snapshot.hasData) {
-      return Text(snapshot.data.title);
+      return Text(snapshot.data!.title);
```

Comment:
If only type flow analysis was smart enough to know `data` can't be null here :)